### PR TITLE
Feature 984 improve releasing + Feature 768 download link latest version

### LINKED ISSUE #768: Provide latest version links at documentation page with refresh actions
## Situation
For automated client updates users would like to have an URL for the latest client release. So they can download binaries from the release directly.

Currently this is not possible, because "latest" release can be a client, a pds or even a server release.

## Wanted
Provide a URL at the documentation pages which gives users a redirect to the corresponding latest xyz release.

## Solution
We provide at our static page (https://mercedes-benz.github.io/sechub/) links to sub pages for each latest versions for different types
e.g.

https://mercedes-benz.github.io/sechub/latest/client.html
https://mercedes-benz.github.io/sechub/latest/server.html
https://mercedes-benz.github.io/sechub/latest/pds.html

and provide refresh / redirect mechanism via 
```
<html>
<head>
   <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
   <meta http-equiv="refresh" content="0; URL=http://example.com/">
   <title>Main Page</title>
</head>
<body></body>
</html>
```

Those pages must be generated at documentation level when dedicated new tags are set - e.g. a `v3.0.0-server`

### LINKED ISSUE #984: improve releasing by updating docs automatically
# Current situation
For each release the generated docs need to be manually
  - downloaded
  - extracted
  - copied to the git branch named `documentation` (only the right files belonging to the release. E.g. sechub-client.html)
  - committed
  - pushed

This eats time and is error-prune (accidentally, docs can be overridden).

# Solution
- Put the officially released documentation into same branch (master) under `docs/released`
- Create a script for running at release time that
  - checks if it's running on master branch (makes no sense on other branches)
  - copies the right files into `docs/released/sechub`
  - `git add` the same
  - `git add` changed image files from `docs/released/images` to `docs/released/sechub/images`
- update Github Actions file `create-releases.yml`
  - call above script
  - commit + push the changes
- Point `https://mercedes-benz.github.io/sechub/` to `docs/released`
- Update release instructions at https://github.com/mercedes-benz/sechub/projects/1

...and enjoy the lean process at next SecHub release

### PR DESCRIPTION
**Pull request by sven-dmlr**
## Summary:
1. releasing improved (#984)
    - official docs moved from branch 'documentation' into main branch('master') under `docs/`
    - 1:1 moved from 'documentation' branch (these files can be skipped in review)
    - updated `docs/index.html`: now relative links
    - added a step into `.github/workflows/create-releases.yml` that automatically adds docs for released products. (This step had to be done formerly). So releasing should go much quicker now.

2. provide fixed links for downloading the latest version of a product (#768)
    - files `client-download.html`, `pds-download.html`, `server-download.html` are created now in gradle documentation run. They contain a forward to the download of the latest released version.
    - Client documentation updated accordingly

closes #768
closes #984

### CODE REVIEW

**Review by haerter-tss — APPROVED** [review]

File: docs/helperscripts/git-add-releasedocs.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+done < <(git tag --points-at HEAD)
+
+# Copy files to destination and stage them for commit
+for file in $FILE_LIST ; do
+  /bin/cp "$SOURCE_DIR/$file" "$DEST_DIR/$file"
+  echo "git add -f \"$DEST_DIR/$file\""
+  git add -f "$DEST_DIR/$file"
+done
+
+# Important: We do no git commit here - so everything the script does, can be undone.
```
Comment (Jeeppler) [inline docs/helperscripts/git-add-releasedocs.sh:74]: Is this comment really suppose to be in the last line?

**Review by Jeeppler — CHANGES_REQUESTED** [review]
See comments.

File: docs/helperscripts/create-download-html.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    ;;
+  *)
+    echo "Product '$product' is not yet implemented."
+    exit 1
+    ;;
+esac
+
+# Create HTML
+cat - <<EOF
+<html>
```
[the remark this answers is not in the document]
Comment (sven-dmlr) [reply docs/helperscripts/create-download-html.sh]: Done. Thanks.

File: docs/helperscripts/git-add-releasedocs.sh [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+cd `dirname $0`/..
+
+BRANCH=`git rev-parse --abbrev-ref HEAD`
+if [ "$BRANCH" != "$GIT_RELEASE_BRANCH" ] ; then
+  echo "$0: This release script is intended to run only on '$GIT_RELEASE_BRANCH' branch. You are on branch '$BRANCH'."
+  echo "Exiting..."
+  exit 1
+fi
+
+# Always update images directory
```
[the remark this answers is not in the document]
Comment (sven-dmlr) [reply docs/helperscripts/git-add-releasedocs.sh]: You're right. Comment is changed.

File: docs/helperscripts/git-add-releasedocs.sh (hunk 1 above)
Comment (sven-dmlr) [reply docs/helperscripts/git-add-releasedocs.sh]: This is be the position in the script where someone would add a `git commit`.
That's why I put the comment there.

**Review by Jeeppler — APPROVED** [review]

**Pull request merged**